platform/darwin/src/MGLLogging_Private.h

```diff
@@ -0,0 +1,20 @@
+#import "MGLLogging.h"
+
+#define mgl_log_info(message, ...)  mgl_log_with_type(MGLLoggingLevelInfo, __PRETTY_FUNCTION__, __LINE__, message, ##__VA_ARGS__)
+#define mgl_log_debug(message, ...) mgl_log_with_type(MGLLoggingLevelDebug, __PRETTY_FUNCTION__, __LINE__, message, ##__VA_ARGS__)
+#define mgl_log_error(message, ...) mgl_log_with_type(MGLLoggingLevelError, __PRETTY_FUNCTION__, __LINE__, message, ##__VA_ARGS__)
+#define mgl_log_fault(message, ...) mgl_log_with_type(MGLLoggingLevelFault, __PRETTY_FUNCTION__, __LINE__, message, ##__VA_ARGS__)
+#define mgl_log_with_type(type, function, line, message, ...) \
+{ \
+    if ([MGLLogging sharedInstance].loggingLevel != MGLLoggingLevelNone && type <= [MGLLogging sharedInstance].loggingLevel) \
+    { \
+        NSString *fmt = [NSString stringWithFormat:(message), ##__VA_ARGS__];  \
```

Comment:
If we're using the default handler, we potentially have 3 `stringWithFormat` before logging the message. Would be nice to avoid that.

One great thing about `os_log` is it defers the formatting - it would be a shame to throw that away - what can we do here?

Reply:
That was a tradeoff for allowing 3rd party logging handlers, the reason behind is that whoever choose to connect their own logger will get the message as we originally intended.